MRML/vtkIGTLToMRMLPolyData.cxx

```diff
@@ -380,10 +380,9 @@ int vtkIGTLToMRMLPolyData::MRMLToIGTL(unsigned long event, vtkMRMLNode* mrmlNode
     
     //------------------------------------------------------------
     // Allocate Status Message Class
-    if (this->OutPolyDataMessage.IsNull())
-      {
-      this->OutPolyDataMessage = igtl::PolyDataMessage::New();
-      }
+    // Overwrite old message because we want to clear arrays from previous
+    // calls that will not be set anew.
+    this->OutPolyDataMessage = igtl::PolyDataMessage::New();
```

Comment:
Sounds good to me.

I can pull @christiana 's change as it is, and then add Clear() call when I merge it to the trunk repository. I suggest we do this first before implementing the new igtl::PolyDataMessage:Clear(), because this could at least address the current issue, and we do not need to wait for the change in OpenIGTLink library, which may take some time.  Also, the current igtl::PolyDataMessage:Clear() may still have some advantage because it does not re-allocate memory for serialized data and all member variables for the class. 
If we end up just re-implementing igtl::PolyDataMessage:Clear(), we will not need to change the OpenIGTLink IF module later.

So, if both @christiana and @lassoan agree, I'll merge this change ASAP, and will start working on the OpenIGTLink library.